Added new Tuya v2 Integration (#19525)
home-assistant/home-assistant.github.io: source/_integrations/tuya.markdown

ha_dhcp: true
---

> The Tuya Home Assistant integration is developed for controlling **Powered by Tuya (PBT)** devices, which is officially maintained by the Tuya Developer Team.

Reviewer: ```suggestion
The Tuya integration allows integrating all Powered by Tuya devices you have added to the Tuya Smart and Tuya Smart Life apps. This integration is officially maintained by Tuya.
```

Author: fixed.

> It applies to smart devices that you have added to the Tuya Smart or Smart Life app.

Reviewer: ```suggestion
```

Author: fixed

{% include integrations/config_flow.md %}

## Supported platforms
- **Climate**: Supports air conditioner, heater, thermostat.
- **Cover**: Supports curtain, curtain switch.
- **Fan**: Supports fan, air purifier, humidifier, dehumidifier.
- **Light**: Supports light, light strip, ambient light, light string, humidifier's light, Ceiling light.
- **Scene**: Supports smart scene.

> - **Switch**: Supports switch, socket, power strip, smart kettle, breaker, pet water feeder, air purifier, diffuser.

Reviewer: ```suggestion
- **Climate**: air conditioner, heater, thermostat.
- **Cover**: curtain, curtain switch.
- **Fan**: fan, air purifier, humidifier, dehumidifier.
- **Light**: light, light strip, ambient light, light string, humidifier's light, Ceiling light.
- **Scene**: smart scene.
- **Switch**: switch, socket, power strip, smart kettle, breaker, pet water feeder, air purifier, diffuser.
```

Author: fixed.

## Quick installation and startup

### Configure on the Tuya IoT Platform

#### Prerequisites

You have created an account for [Tuya Smart or Smart Life app](https://developer.tuya.com/en/docs/iot/tuya-smart-app-smart-life-app-advantages?id=K989rqa49rluq#title-1-Download).

#### Create a project

1. Log in to the [Tuya IoT Platform](https://iot.tuya.com/).
2. In the left navigation bar, click **Cloud** > **Projects**.
3. On the page that appears, click **Create**.

> 4. In the **Create Project** dialog box, configure **Project Name**, **Description**, **Industry**, and **Availability Zone**. In the **Development Method** field, select **Smart Home PaaS** from the dropdown list.
> 	> **Note:** Select all available zones.

Reviewer: ```suggestion
4. In the **Create Project** dialog box, configure **Project Name**, **Description**, **Industry**, and **Availability Zone**. In the **Development Method** field, select **Smart Home PaaS** from the dropdown list. Select all available zones.
```

Author: fixed.

> ![](../images/integrations/tuya/image_001.png)

Reviewer: ```suggestion
![](/images/integrations/tuya/image_001.png)
```

Author: fixed

	
5. Click **Create** to continue project configuration.
6. In **Configuration Wizard**, select **Device status notification** API.
   > **Note:** **Smart Home PaaS** API has been selected as default.
   

> ![](../images/integrations/tuya/image_002.png)

Reviewer: ```suggestion
![](/images/integrations/tuya/image_002.png)
```

Author: fixed

7. Click **Authorize**.

#### Get authorization key

Click the created project to enter the **Project Overview** page and get the **Authorization Key** used to make API calls.

> ![](../images/integrations/tuya/image_003.png)

Reviewer: ```suggestion
![](/images/integrations/tuya/image_003.png)
```

Author: fixed

#### Link devices by app account

Link devices by your app account and copy the **Device ID** in the **Device List** to the clipboard for specifying `device_id` later.

**1.** Navigate to the **Devices** tab.

**2.** Click **Link Tuya App Account** > **Add App Account**.

> ![](../images/integrations/tuya/image_004.png)

Reviewer: ```suggestion

![](/images/integrations/tuya/image_004.png)
```

Author: fixed

**3.** Scan the QR code that appears using the **Tuya Smart** app or **Smart Life** app.

> ![](../images/integrations/tuya/image_005.png)

Reviewer: ```suggestion

![](/images/integrations/tuya/image_005.png)
```

Author: fixed

**4.** Click **Confirm** on your **Tuya Smart** or **Smart Life** app.

**5.** Navigate to the **Device List** tab. You can find the devices that have been added through your **Tuya Smart** or **Smart Life** account.   

> ![](../images/integrations/tuya/image_006.png)

Reviewer: ```suggestion

![](/images/integrations/tuya/image_006.png)
```

Author: fixed

> </details>

Reviewer: ```suggestion
```

Author: removed.

### Install the Tuya Home Assistant integration

#### Install Home Assistant

Please refer to the [Home Assistant Official Installation](https://www.home-assistant.io/installation/) documentation to install **Home Assistant Core**.

You can also get help from [Set up Home Assistant Development Environment on Raspberry Pi](https://developer.tuya.com/en/demo/setuphomeassistantdevenv/) in Tuya Developer Demo Center.

#### Install the Tuya integration

In **Configuration** > **Integrations** > **ADD INTEGRATION**, search for the keyword **Tuya** and select it for installation.

> ![](../images/integrations/tuya/image_007.png)

Reviewer: This can all be removed. People are in the HA docs.

Author: OK, we will keep the "Install the Tuya integration" section.

#### Enter your Tuya credential

In the Tuya Integration window, select **Smart Home PaaS** and click **SUBMIT**.

![](../images/integrations/tuya/image_008.png)

See the following table and enter your Tuya credential.

| Field | Description |
| ------- | -------- |
| Access ID and Access Secret| Go to your cloud project on [Tuya IoT Platform](https://iot.tuya.com/). Find the **Access ID** and **Access Secret** on the **Project Overview** tab.|
| Mobile App | Must select the one you used to link devices on the Tuya IoT Platform. |
| Country Code | The country you select on logging in to Tuya Smart or Smart Life app.|
| Account | Tuya Smart or Smart Life app account. |
| Password | The password of your app account. |

> > **Note**: The app mentioned in the table must be the one you used to link devices on the Tuya IoT Platform.
> </details>

Reviewer: ```suggestion

<p class='note'>The app mentioned in the table must be the one you used to link devices on the Tuya IoT Platform.</p>
```

Author: fixed.
